src/transform/primitives.js

```diff
@@ -12,17 +12,14 @@ exports.primitives = {
     return null
   },
   booleanify: function(val) {
-    if (val === 'false') {
-      return false
-    }
-
-    // null or undefined values will return null
-    if (val == null) {
-      return null
-    }
-
     // all other values will be converted to boolean by their truthy or falsy value
-    // ie: 0, 0n, '', ``, "", NaN, false are falsy and everything else including -1, {}, [] are truthy
-    return !!val
+    // ie: 0, 0n, '', ``, "", NaN, and false are falsy and everything else including -1, {}, and [] are truthy
+    return Boolean(val)
+  },
+  numberify: function(val) {
+    return Number(val)
+  },
+  stringify: function(val) {
+    return String(val)
```

Comment:
I wonder if we shouldnt maybe be doing `JSON.stringify` here. This way an entire object/array can be converted into a string format to be parsed again at a later stage.

Reply:
Sounds good :100: I'm just going to do a check for the data type coming in before doing JSON.stringify() as actual strings coming in will be returned with an extra set of inverted commas in they go through JSON.stringify().